bedita-app/vendors/shells/tasks/cleanup_data.php

```diff
@@ -0,0 +1,161 @@
+ */
+class CleanupDataTask extends BeditaBaseShell {
+    
+    /**
+     * @inheritDoc
+     */
+    public $uses = ['BEObject'];
+
+    protected $truncateTables = ['event_logs', 'mail_jobs', 'mail_logs'];
+
+    protected $cleanupTables = ['history', 'versions'];
+
+    public function help() {
+        $this->hr();
+        $this->out('cleanup data script shell usage:');
+        $this->out('');
+        $this->out('./cake.sh bedita cleanupData // truncate event_logs, mail_jobs, mail_logs, clean old data from history and versions');
+        $this->out('./cake.sh bedita cleanupData -ld <limit date yyyy-MM-dd> // use custom limit date for clean');
+        $this->out('./cake.sh bedita cleanupData -ni // no interactive mode');
+        $this->out('./cake.sh bedita cleanupData -tt <comma separated table names> // truncate event_logs, mail_jobs, mail_logs + more tables');
+        $this->out('./cake.sh bedita cleanupData help // show this help');
+        $this->out('');
+    }
+
+    /**
+     * Tables to clean (truncate or delete data):
+     * - event_logs (truncate)
+     * - mail_jobs (truncate, if no messages pending)
+     * - mail_logs (truncate)
+     * - history (cleanup: mantain only "CY-1 / CY" data (CY is the Current Year))
+     * - versions (cleanup: mantain only "CY-1 / CY" data (CY is the Current Year))
+     *
+     * @return void
+     */
+    public function execute()
+    {
+        $this->hr();
+        $this->out('BEdita Data Cleanup');
+        $this->out('--- count data before cleaning ---');
+        $tables = $this->truncateTables;
+        if (isset($this->params['tt'])) {
+            $tables = array_merge($tables, explode(',', $this->params['tt']));
+        }
+        $countTables = array_merge($this->cleanupTables, $tables);
+        sort($countTables);
+        $limitDate = date('Y')-2 . '-12-31 23:59:59';
+		if (isset($this->params['ld'])) {
+			$limitDate = $this->params['ld'] . ' 23:59:59';
+        }
+        foreach ($countTables as $tableName) {
+            if (in_array($tableName, $this->cleanupTables)) {
+                $counts[$tableName] = $this->countRecords($tableName, $limitDate);
+            } else {
+                $counts[$tableName] = $this->countRecords($tableName);
+            }
+        }
+        // check if mail_jobs can be truncated
+        if ($counts['mail_jobs'] > 0) {
+            $query = 'SELECT COUNT(*) AS n FROM mail_jobs WHERE status NOT IN (\'sent\', \'failed\')';
+            $result = $this->BEObject->query($query);
+            $pending = $result[0][0]['n'];
+            $this->out(sprintf('%s => %s', $query, $pending));
+            if ($pending > 0) {
+                $this->out('mail_jobs won\'t be truncated, because there are pending jobs. cleanup instead');
+                if (($key = array_search('mail_jobs', $tables)) !== false) {
+                    unset($tables[$key]);
+                }
+                $this->cleanupTables[] = 'mail_jobs';
+            }    
```

Comment:
Since the table is not locked, there's no guarantee that a pending job cannot be created between this `SELECT` and the `TRUNCATE TABLE`.

I'd just go with the cleanup always.